src/super_gradients/training/metrics/segmentation_metrics.py

```diff
@@ -174,27 +205,73 @@ def __init__(self, ignore_label=-100, dist_sync_on_step=False, metrics_args_prep
 
     def update(self, preds: torch.Tensor, target: torch.Tensor):
         predict, target = self.metrics_args_prep_fn(preds, target)
+        labeled_mask = self._handle_multiple_ignored_inds(target)
 
-        labeled_mask = target.ne(self.ignore_label)
         pixel_labeled = torch.sum(labeled_mask)
         pixel_correct = torch.sum((predict == target) * labeled_mask)
         self.total_correct += pixel_correct
         self.total_label += pixel_labeled
 
+    def _handle_multiple_ignored_inds(self, target):
+        if isinstance(self.ignore_label, list):
```

Comment:
Just a slight optimization, which may be faster:
labeled_mask = torch.ones_like(target)
for ignored_label:
   labeled_mask.masked_fill(target.eq(ignored_label), 0)

This avoids creating temporary buffers and should be a bit faster.

Reply:
might be missing something, but don't see why it would faster (since masked_fill is not in place).
Nonetheless I updated it.